include/mbedtls/build_info.h

```diff
@@ -112,6 +112,14 @@
 #undef MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED
 #undef MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED
 #undef MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED
+#endif
+
+#if !defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED) && \
+    !defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED)
+#undef MBEDTLS_SSL_EARLY_DATA
+#endif
+
+#if !defined(MBEDTLS_SSL_SESSION_TICKETS)
```

Comment:
It's not covered, this check is to fix the CI failure.  see comment https://github.com/Mbed-TLS/mbedtls/pull/6486#discussion_r1007746949

Reply:
What's the CI failure? is it not covered by configs.py ?

Reply:
Can I learn which part of config.py will cover this?
I just combined several commits into the first commit of this PR, and before I add this check in build_info.h,  the tls13 related tests will report the error in check_config.h as **"MBEDTLS_SSL_EARLY_DATA  defined, but not all prerequisites"**

Reply:
```
Requires: MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED or MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED
```
with above sentence , `configs.py full` should set them. If `configs.py full` does not set them, that's an exception case. 

And for the case you said, I prefer enable MBEDTLS_SSL_EARLY_DATA MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED/MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED together.  

This change will waste developer time. If developer set `EARLY_DATA` without `SOME_PSK` , EARLY_DATA feature will disappear without notice. Developer have to spend time on finding the root cause.

Reply:
> ```
> Requires: MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED or MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED
> ```
> 
> with above sentence , `configs.py full` should set them. If `configs.py full` does not set them, that's an exception case.

Some components will not `configs.py full`.
> 
> And for the case you said, I prefer enable MBEDTLS_SSL_EARLY_DATA MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED/MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED together.
> 
> This change will waste developer time. If developer set `EARLY_DATA` without `SOME_PSK` , EARLY_DATA feature will disappear without notice. Developer have to spend time on finding the root cause.

I am not sure why waste developer time. Currently, we will enforce the binding implicitly, users have no need to know it, if no some_psk defined,  EARLY_DATA feature should dissapeared, cause it can't work without some_psk.  See comment https://github.com/Mbed-TLS/mbedtls/pull/6486#discussion_r1017647268

Reply:
> Some components will not configs.py full.

If not use configs.py full , add it one by one.

> I am not sure why waste developer time. 

**EARLY_DATA feature will disappear without notice.**  Please read full sentence.

> if no some_psk defined, EARLY_DATA feature should dissapeared, cause it can't work without some_psk. 

Add them one by one is better than disable it without NOTICE.